### ISSUE TITLE
Using Intel iGPU for OpenVINO model inferencing

### ISSUE THREAD

**Issue by PythoneerSamurai** [ultralytics/ultralytics#20143]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Greetings.
Recently I exported my YOLOv10 object detection model to the OpenVINO format using the export function in the Ultralytics YOLO class.
The export went smoothly, and the model showed significant improvements in inference time.
I want to know how I can use my Intel Iris Xe 80 EU iGPU for carrying model inference.
I know how to specify the device to be iGPU when compiling the model using the Intel OpenVINO Python library, but am not sure how to do it for a YOLO model.

My laptop's specs:
Intel Core i5 1235U (10 cores, 12 threads)
Intel Iris XE 80EU iGPU
16 GB DDR4 3200 MHz RAM
250 GB M.2 NVME

Just in case this helps, the model I'm trying to convert is uploaded on my [Kaggle](https://www.kaggle.com/models/pythonistasamurai/yolov8x_v10x_tennis_analysis_models)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @PythoneerSamurai, thank you for your interest in Ultralytics 🚀! We appreciate you sharing your detailed question and system specs—it helps us understand your query better.

We recommend checking the [Docs](https://docs.ultralytics.com/) for guidance on export and inference, including details on [exporting models](https://docs.ultralytics.com/modes/export/) and [OpenVINO](https://docs.ultralytics.com/modes/export/#openvino). You might find helpful insights for specifying devices during inference.

If this is a 🐛 Bug Report or an enhancement request, please also include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug or better understand the issue.

For OpenVINO-specific inference on Intel iGPUs, you may need to configure the device explicitly within your script. If you are using the Ultralytics YOLO export functionality, we encourage you to share any errors or logs you encounter during the process to help us assist you more effectively.

Join the Ultralytics community for further discussion and support:
- Engage in real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- Explore detailed discussions on [Discourse](https://community.ultralytics.com/)
- Share knowledge on our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Ensure your environment is up to date by upgrading to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following verified environments, which include all dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but don't worry—an Ultralytics engineer will review your issue and provide additional assistance as soon as possible. 😊

**Comment by PythoneerSamurai** [comment]
Okay, so I checked that when I carried inference using my exported model my iGPU's Render/3D engine was being used close to a hundred, while my CPU was not being used much, could this mean that my iGPU is being used by default, if so how do I use my CPU for inference?

**Comment by Y-T-G** [comment]
It uses GPU if available, otherwise CPU.

**Comment by PythoneerSamurai** [comment]
> It uses GPU if available, otherwise CPU.

Yeah I found the solution to my problem.
All I had to do was to pass the .xml file generated by the YOLO export function to the OpenVINO compile_model function, then I could force the compiled model to use the device I specified.

**Comment by glenn-jocher** [comment]
I'm glad you found a solution! Yes, when you load the model with OpenVINO's native API using the .xml file, you gain full control over device selection. This approach gives you the flexibility to explicitly choose between CPU, GPU, or other supported Intel hardware.

For others who might need this: OpenVINO supports various device targets including "CPU", "GPU", "AUTO", and others. You can specify these when compiling the model:

```python
import openvino as ov
core = ov.Core()
model = core.read_model("yolov10_openvino_model/yolov10.xml")
compiled_model = core.compile_model(model, "CPU")  # Force CPU usage
```

Thanks for sharing your solution - it's helpful for the community!

**Comment by PythoneerSamurai** [comment]
> I'm glad you found a solution! Yes, when you load the model with OpenVINO's native API using the .xml file, you gain full control over device selection. This approach gives you the flexibility to explicitly choose between CPU, GPU, or other supported Intel hardware.
> 
> For others who might need this: OpenVINO supports various device targets including "CPU", "GPU", "AUTO", and others. You can specify these when compiling the model:
> 
> ```python
> import openvino as ov
> core = ov.Core()
> model = core.read_model("yolov10_openvino_model/yolov10.xml")
> compiled_model = core.compile_model(model, "CPU")  # Force CPU usage
> ```
> 
> Thanks for sharing your solution - it's helpful for the community!

Much appreciated 😊

**Comment by glenn-jocher** [comment]
Great to hear OpenVINO's working well for you! For anyone interested in further optimizing performance, OpenVINO also offers performance hints for different scenarios:

```python
import openvino.properties.hint as hints

# For latency-sensitive applications
config = {hints.performance_mode: hints.PerformanceMode.LATENCY}
compiled_model = core.compile_model(model, "GPU", config)

# For throughput-oriented workloads
config = {hints.performance_mode: hints.PerformanceMode.THROUGHPUT}
compiled_model = core.compile_model(model, "GPU", config)
```

These hints can help fine-tune performance based on your specific use case without needing to manually configure device-specific parameters.
